packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -216,10 +236,17 @@ export class Metric implements IMetric {
     if (periodSec !== 1 && periodSec !== 5 && periodSec !== 10 && periodSec !== 30 && periodSec % 60 !== 0) {
       throw new Error(`'period' must be 1, 5, 10, 30, or a multiple of 60 seconds, received ${periodSec}`);
     }
+    if (props.dimensions && props.dimensionsMap) {
+      throw new Error('Using both the \'dimensions\' and \'dimensionsMap\' properties is not supported. Use only \'dimensionsMap.\'');
+    }
     if (props.dimensions) {
       this.validateDimensions(props.dimensions);
     }
     this.dimensions = props.dimensions;
+    if (props.dimensionsMap) {
+      this.validateDimensionsMap(props.dimensionsMap);
+    }
+    this.dimensionsMap = props.dimensionsMap;
```

Comment:
Thinking about this some more. I would probably try to have a single field that represents both `dimensions` and `dimensionsMap`, and then turn the current `dimensions` field to a getter function, that returns `this.dimensionsMap` - I _think_ the types should work out there.

Reply:
Talked offline with @skinny85 and decided to keep the existing `dimensions` field, and only use the new type for the `dimensionsMap` constructor prop. If the user supplies either `dimensionsMap` or `dimensions` in the constructor, they will both get assigned to the `dimensions` field (with the new prop, `dimensionsMap` taking precedence).